bindings/matrix-sdk-ffi/src/room_directory_search.rs

```diff
@@ -29,12 +29,14 @@ pub enum PublicRoomJoinRule {
     Knock,
 }
 
-impl PublicRoomJoinRule {
-    fn convert(value: ruma::directory::PublicRoomJoinRule) -> Option<Self> {
+impl TryFrom<ruma::directory::PublicRoomJoinRule> for PublicRoomJoinRule {
+    type Error = ();
+
+    fn try_from(value: ruma::directory::PublicRoomJoinRule) -> Result<Self, Self::Error> {
         match value {
-            ruma::directory::PublicRoomJoinRule::Public => Some(Self::Public),
-            ruma::directory::PublicRoomJoinRule::Knock => Some(Self::Knock),
-            _ => None,
+            ruma::directory::PublicRoomJoinRule::Public => Ok(Self::Public),
+            ruma::directory::PublicRoomJoinRule::Knock => Ok(Self::Knock),
+            _ => Err(()),
```

Comment:
```suggestion
            rule => Err(format!("unsupported join rule: {rule:?}")),
```

Also, is it something we want on the long-term? Don't we want to support all join rules here?

Reply:
We only care about public since we are not even going to support knocking, so any custom rule is not needed... for now...